coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala

```diff
@@ -0,0 +1,73 @@
+package filodb.coordinator.queryplanner
+
+import filodb.core.query.QueryContext
+import filodb.query.{BinaryJoin, LabelValues, LogicalPlan, SeriesKeysByFilters, SetOperator}
+import filodb.query.exec._
+
+/**
+  * SinglePartitionPlanner is responsible for planning in situations where time series data is
+  * distributed across multiple clusters.
+  *
+  * @param planners map of clusters names in the local partition to their Planner objects
+  * @param plannerSelector a function that selects the planner name given the metric name
+  *
+  */
+class SinglePartitionPlanner(planners: Map[String, QueryPlanner], plannerSelector: String => String,
+                             datasetMetricColumn: String)
+  extends QueryPlanner {
+
+  def materialize(logicalPlan: LogicalPlan, qContext: QueryContext): ExecPlan = {
+
+    logicalPlan match {
+      case lp: BinaryJoin          => materializeBinaryJoin(lp, qContext)
+      case lp: LabelValues         => materializeLabelValues(lp, qContext)
+      case lp: SeriesKeysByFilters => materializeSeriesKeysFilters(lp, qContext)
+      case _                       => materializeSimpleQuery(logicalPlan, qContext)
+
+    }
+  }
+
+  private def materializeSimpleQuery(logicalPlan: LogicalPlan, qContext: QueryContext): ExecPlan = {
+
+    planners.get(plannerSelector(LogicalPlanUtils.getMetricName(logicalPlan).head)).get.
+      materialize(logicalPlan, qContext)
+  }
+
+  private def materializeBinaryJoin(logicalPlan: BinaryJoin, qContext: QueryContext): ExecPlan = {
+
+    val lhsMetrics = LogicalPlanUtils.getMetricName(logicalPlan.lhs)
+    val rhsMetrics = LogicalPlanUtils.getMetricName(logicalPlan.rhs)
+    val lhsPlanner = planners.get(plannerSelector(lhsMetrics.head)).get
+    val rhsPlanner = planners.get(plannerSelector(rhsMetrics.head)).get
```

Comment:
Are LHS and RHS both guaranteed to have metric name ?

Reply:
Yes not necessary. Made changes as discussed above